tests/ui/resolve/issue-112831.rs

```diff
@@ -9,7 +8,7 @@ use zeroable::*;
 
 mod pod {
     use super::*;
-    pub trait Pod: Zeroable {}
+    pub trait Pod: Zeroable {} //~ ERROR expected trait, found derive macro `Zeroable`
```

Comment:
It seems like a correct fix. Do we need to add a lint for this?

Reply:
This fix is not correct because we need to locate ﻿`(Mod(root::pod), Key(Zeroable, TypeNS))` instead of `﻿(Mod(root::pod), Key(Zeroable, Macro))`. Therefore, this case should pass the check.